ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +360,709 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+/**
+ * Update the description of a secret
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}secretName(String [1~64]) : The name of the secret. eg. 'secretName01'
+ *          ==> {o}description(String [1~4096]) : The description of the secret. eg. 'desc01'
+ * @returns
+ */
+const updateSecretDesc = async (res, appid, DB, payload) => {
+    //get and check param in payload
+    let secretName = getParam_String(payload, 'secretName')
+    let description = getParam_String(payload, 'description')
+    if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+        res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+        return
+    }
+    if (!checkStringParam(description, false, DESCRIPTION_LENGTH_MAX)) {
+        res.send(_result(400, `description must be string and length not more than ${DESCRIPTION_LENGTH_MAX}`))
+        return
+    }
+
+    //Query the description through secret name and update the description
+    const secret_name_query = {
+        selector: {
+            appid,
+            secretName
+        },
+        fields: [
+            '_id',
+            '_rev',
+            'appid',
+            'secretName',
+            'encryptionKeyId',
+            'description',
+            'createTime',
+            'deleteTime',
+            'plannedDeleteTime',
+            'rotationInterval',
+            'lastRotationDate',
+            'nextRotationDate'
+        ],
+        limit: 1,
+    }
+    await DB.partitionedFind('secret_metadata', secret_name_query)
+        .then((secret_metadata_res) => {
+            if (secret_metadata_res.docs.length > 0) {
+                secret_metadata_res.docs[0].description = description
+                DB.insert(secret_metadata_res.docs[0])
```

Comment:
Is there any API to update the description only? Seems the current logic is to read all items and update them all together. @393~@404

Reply:
CouchDB does not have a separate field update. All fields must be updated.